tiledb/sm/cpp_api/capi_string.h

```diff
@@ -0,0 +1,108 @@
+/**
+ * @file   capi_string.h
+ *
+ * @section LICENSE
+ *
+ * The MIT License
+ *
+ * @copyright Copyright (c) 2017-2023 TileDB, Inc.
+ *
+ * Permission is hereby granted, free of charge, to any person obtaining a copy
+ * of this software and associated documentation files (the "Software"), to deal
+ * in the Software without restriction, including without limitation the rights
+ * to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+ * copies of the Software, and to permit persons to whom the Software is
+ * furnished to do so, subject to the following conditions:
+ *
+ * The above copyright notice and this permission notice shall be included in
+ * all copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+ * THE SOFTWARE.
+ *
+ * @section DESCRIPTION
+ *
+ * This file declares the C++ API for the TileDB CAPIString object.
+ */
+
+#ifndef TILEDB_CPP_API_CAPI_STRING_H
+#define TILEDB_CPP_API_CAPI_STRING_H
+
+#include <cassert>
+#include <optional>
+#include <stdexcept>
+#include "tiledb.h"
+
+namespace tiledb::impl {
+
+/**
+ * Manages the lifetime of a tiledb_string_t* handle and provides operations on
+ * it.
+ */
+class CAPIString {
+ public:
+  CAPIString(tiledb_string_t*&& handle) noexcept {
+    string_ = handle;
+    handle = nullptr;
```

Comment:
If you want move semantics, then you want a handle to the handle, that is, `CAPIString(tiledb_string_t** handle)`. Then you want this statement to be `*handle = nullptr`.

Reply:
Switched to `tiledb_string_t**`.